src/daft-scan/src/python.rs

```diff
@@ -1,46 +1,119 @@
 use pyo3::prelude::*;
 
 pub mod pylib {
+    use std::sync::Arc;
+
     use pyo3::prelude::*;
-    use std::str::FromStr;
 
     use daft_core::python::schema::PySchema;
 
     use pyo3::pyclass;
+    use serde::{Deserialize, Serialize};
 
     use crate::anonymous::AnonymousScanOperator;
-    use crate::FileType;
-    use crate::ScanOperatorRef;
+    use crate::file_format::PyFileFormatConfig;
+    use crate::storage_config::PyStorageConfig;
+    use crate::{ScanOperatorRef, ScanTask, ScanTaskBatch};
 
     #[pyclass(module = "daft.daft", frozen)]
-    pub(crate) struct ScanOperator {
+    #[derive(Debug, Clone)]
+    pub struct ScanOperatorHandle {
         scan_op: ScanOperatorRef,
     }
 
     #[pymethods]
-    impl ScanOperator {
+    impl ScanOperatorHandle {
         pub fn __repr__(&self) -> PyResult<String> {
             Ok(format!("{}", self.scan_op))
         }
 
         #[staticmethod]
         pub fn anonymous_scan(
-            schema: PySchema,
-            file_type: &str,
             files: Vec<String>,
+            schema: PySchema,
+            file_format_config: PyFileFormatConfig,
+            storage_config: PyStorageConfig,
         ) -> PyResult<Self> {
             let schema = schema.schema;
-            let operator = Box::new(AnonymousScanOperator::new(
-                schema,
-                FileType::from_str(file_type)?,
+            let operator = Arc::new(AnonymousScanOperator::new(
                 files,
+                schema,
+                file_format_config.into(),
+                storage_config.into(),
             ));
-            Ok(ScanOperator { scan_op: operator })
+            Ok(ScanOperatorHandle { scan_op: operator })
+        }
+    }
+
+    impl From<ScanOperatorRef> for ScanOperatorHandle {
+        fn from(value: ScanOperatorRef) -> Self {
+            Self { scan_op: value }
+        }
+    }
+
+    impl From<ScanOperatorHandle> for ScanOperatorRef {
+        fn from(value: ScanOperatorHandle) -> Self {
+            value.scan_op
+        }
+    }
+
+    #[pyclass(module = "daft.daft", name = "ScanTask", frozen)]
+    #[derive(Debug, Clone, Serialize, Deserialize)]
+    pub struct PyScanTask(Arc<ScanTask>);
```

Comment:
Can we merge in these changes for daft scan python before the rest of the PR? I was planning on implementing the same logic in my PR to create scan tasks from iceberg. Would be great not to diverge

Reply:
Iceberg branch has been rebased